encoding/kzg/verifier/verifier.go

```diff
@@ -70,21 +70,24 @@ func NewVerifier(config *kzg.KzgConfig, loadG2Points bool) (*Verifier, error) {
 			return nil, err
 		}
 	} else {
-		if len(config.G2PowerOf2Path) == 0 {
-			return nil, errors.New("G2PowerOf2Path is empty. However, object needs to load G2Points")
+		if len(config.G2PowerOf2Path) == 0 && len(config.G2Path) == 0 {
```

Comment:
The config.G2Path isn't used below, although it's checking it here.
What if the control flow is `if loadG2Points || len(config.G2PowerOf2Path) == 0 { ... }` above? This entire block won't need to change

Reply:
It is used to ensure backward compatibility for operators who uses the entire G2 points, though very unlikely. In the last PR #472 , node is required to read the powerOf2 file at start. However, it prevents the legacy usecase when the node is accessing the entire points.

Reply:
node sets loadG2Points to be false.

Reply:
I think my suggestion has the same logic, but simpler code?